### DISCUSSION TITLE
HELP! how can i change chunkid where in webpackjsonp

### DISCUSSION THREAD

**Discussion by Taoja** [webpack/webpack#15816] [Q&A] [answered]
i wanna try change chunkid like this:
before:
```js
(window["webpackJsonp"] = window["webpackJsonp"] || []).push([["investment/self/detail/index"],{
...
```
changed:
```js
(window["webpackJsonp"] = window["webpackJsonp"] || []).push([["investment/self/detail/index_hash"],{
...
```

i tried write a custom plugin but got failed
```js
compilation.hooks.optimizeChunkIds.tap('chunkIdHash', chunks => {
    chunks.forEach(chunk =>  chunk.id = `${chunk.id}_${hash}`)
})
```
used webpack 4+

**Comment by alexander-akait** [comment, marked as answer]
You can create custom plugin for this, here example https://github.com/webpack/webpack/blob/webpack-4/lib/NamedChunksPlugin.js

**Reply by Taoja** [reply]
> You can create custom plugin for this, here example https://github.com/webpack/webpack/blob/webpack-4/lib/NamedChunksPlugin.js

thx alot， i did it
